tests/ctst/steps/bucket-policies/common.ts

```diff
@@ -0,0 +1,475 @@
+                `arn:aws:s3:::${bucketName}badname/*`,
+            ];
+        }
+        if (action.excludePermissionOnBucketObjects) {
+            resources.pop();
+        }
+        if (action.excludePermissionsOnBucket) {
+            resources.shift();
+        }
+    } else {
+        authzConfiguration.Resource = AuthorizationType.NO_RESOURCE;
+        return;
+    }
+    this.addToSaved('authzConfiguration', authzConfiguration);
+    const basePolicy = {
+        Version: '2012-10-17',
+        Statement: [
+            {
+                Effect: effect,
+                Action: action.permissions,
+                Resource: resources,
+            },
+        ],
+    };
+    if (isAllow === 'ALLOW+DENY') {
+        basePolicy.Statement.push({
+            Effect: effect === AuthorizationType.ALLOW ? AuthorizationType.DENY : AuthorizationType.ALLOW,
+            Action: action.permissions,
+            Resource: resources,
+        });
+    }
+    const createdPolicy = await IAM.createPolicy({
+        policyDocument: JSON.stringify(basePolicy),
+        policyName: `policyforauthz-${Utils.randomString()}`,
+    });
+    const policyArn = extractPropertyFromResults(createdPolicy, 'Policy', 'Arn') as string;
+
+    const identityType = this.getSaved<string>('identityType') as EntityType;
+    if (identityType === EntityType.ASSUME_ROLE_USER
+        || identityType === EntityType.ASSUME_ROLE_USER_CROSS_ACCOUNT
+        || identityType === EntityType.DATA_CONSUMER) {
+        const result = await IAM.attachRolePolicy({
+            policyArn,
+            roleName: this.getSaved<string>('identityName'),
+        });
+        assert.ifError(result.stderr || result.err);
+    }
+    if (identityType === EntityType.IAM_USER) {
+        const result = await IAM.attachUserPolicy({
+            policyArn,
+            userName: this.getSaved<string>('identityName'),
+        });
+        assert.ifError(result.stderr || result.err);
+    }
+});
+
+Given('a policy granting full access to the objects and read access to the bucket', async function (this: Zenko) {
+    this.setAuthMode('base_account');
+    const authzConfiguration: AuthorizationConfiguration = {
+        Identity: this.getSaved<AuthorizationConfiguration>('authzConfiguration')?.Identity
+            || AuthorizationType.NO_RESOURCE,
+        Resource: AuthorizationType.ALLOW,
+    };
+    const action = this.getSaved<ActionPermissionsType>('currentAction');
+    if (writeOperationsOnABucket.includes(action.action)) {
+        authzConfiguration.Resource = AuthorizationType.DENY;
+    }
+    this.addToSaved('authzConfiguration', authzConfiguration);
+    const bucketName = this.getSaved<string>('bucketName');
+    const identityType = this.getSaved<string>('identityType') as EntityType;
+    const currentIdentityArn = this.getSaved<string>('identityArn');
+    const accountId = currentIdentityArn.split(':')[4];
+    let principal = `arn:aws:iam::${accountId}:root`;
```

Comment:
I don't remember us using root Principal for BPs in previous tests, can you refresh me on this?